test/index.js

```diff
@@ -675,11 +675,19 @@ test ('equals', function() {
   eq (Z.equals (Math.sin, Math.cos), false);
   eq (Z.equals (Identity (Identity (Identity (0))), Identity (Identity (Identity (0)))), true);
   eq (Z.equals (Identity (Identity (Identity (0))), Identity (Identity (Identity (1)))), false);
-  eq (Z.equals (Useless, Useless), false);
   eq (Z.equals (Array.prototype, Array.prototype), true);
   eq (Z.equals (Nothing.constructor, Maybe), true);
   eq (Z.equals ((Just (0)).constructor, Maybe), true);
-  eq (Z.equals (Lazy$of (0), Lazy$of (0)), false);
+
+  assert.throws (
+    () => { Z.equals (Useless, Useless); },
+    new TypeError ("Cannot read property 'bind' of null")
```

Comment:
Did you consider detecting this case and providing a more informative error message? Mentioning `fantasy-land/equals` here would be helpful.

Reply:
Yes. I intended to make that change in a follow-up PR. I prefer to consider the perf PR first though, because the code we'd need to touch gets reworked there.

Reply:
The reason this test case is listed is because it used to be a valid input. For most other functions, the garbage paths are not tested. Perhaps I should remove this test case altogether.

Reply:
Actually it's silly to test "unspecified behaviour", because that makes it specified. :P

Reply:
I've opened #152 to track your suggestion. I'll also remove this test case from the PR.